modules/navier_stokes/src/userobjects/NSFVPressurePin.C

```diff
@@ -0,0 +1,127 @@
+
+  return params;
+}
+
+NSFVPressurePin::NSFVPressurePin(const InputParameters & params)
+  : GeneralUserObject(params),
+    TaggingInterface(this),
+    BlockRestrictable(this),
+    NonADFunctorInterface(this),
+    _mesh(UserObject::_subproblem.mesh().getMesh()),
+    _p(dynamic_cast<INSFVPressureVariable *>(
+        &UserObject::_subproblem.getVariable(0, getParam<NonlinearVariableName>("variable")))),
+    _p0(getPostprocessorValue("phi0")),
+    _pressure_pin_type(getParam<MooseEnum>("pin_type")),
+    _pressure_pin_point(_pressure_pin_type == "point-value" ? getParam<Point>("point")
+                                                            : Point(0, 0, 0)),
+    _current_pressure_average(
+        _pressure_pin_type == "average" ? &getPostprocessorValue("pressure_average") : nullptr),
+    _sys(*getCheckedPointerParam<SystemBase *>("_sys"))
+{
+  if (!_p)
+    paramError(NS::pressure, "the pressure must be a INSFVPressureVariable.");
+  // Check execute_on of the postprocessor
+  if (_pressure_pin_type == "average" &&
+      !_fe_problem.getUserObjectBase(getParam<PostprocessorName>("pressure_average"))
+           .getExecuteOnEnum()
+           .contains(getExecuteOnEnum()))
+    paramError("pressure_average",
+               "Pressure average postprocessor must include the pin execute_on flags");
+}
+
+void
+NSFVPressurePin::execute()
+{
+  // For average pressure interpolation, we expect a checkerboard.
+  // We can smooth it here by average each value with the average of several of its neighbors
+  // This will change the value of the pressure pin, but if we are careful it will not change the
+  // average (provided by the postprocessor)
+
+  // Get the value of the pin
+  Real pin_value = 0;
+  if (_pressure_pin_type == "point-value")
+  {
+    Real point_value = _sys.system().point_value(_p->number(), _pressure_pin_point, false);
+
+    /**
+     * If we get exactly zero, we don't know if the locator couldn't find an element, or
+     * if the solution is truly zero, more checking is needed.
+     */
+    if (MooseUtils::absoluteFuzzyEqual(point_value, 0.0))
+    {
+      auto pl = _mesh.sub_point_locator();
+      pl->enable_out_of_mesh_mode();
+
+      auto * elem = (*pl)(_pressure_pin_point);
+      auto elem_id = elem ? elem->id() : DofObject::invalid_id;
+      gatherMin(elem_id);
+
+      if (elem_id == DofObject::invalid_id)
+        mooseError("No element to gather point pressure from located at ", _pressure_pin_point);
+      // Default at construction
+      pl->disable_out_of_mesh_mode();
+    }
+
+    pin_value = _p0 - point_value;
+  }
+  else
+    pin_value = _p0 - *_current_pressure_average;
+
+  // Offset the entire pressure vector by the value of the pin
+  NumericVector<Number> & sln = _sys.solution();
+  std::set<dof_id_type> local_dofs;
+  _sys.system().local_dof_indices(_p->number(), local_dofs);
+  for (const auto & dof : local_dofs)
+    sln.add(dof, pin_value);
+  sln.close();
```

Comment:
```suggestion
  sln.close();
  _sys.system().update();
```
To be safe I would ensure the changes in the parallel vector get into the ghosted solution vector, since that is what is read by most objects